drivers/watchdog/wdt_nrfx.c

```diff
@@ -71,29 +42,20 @@ static int wdt_nrf_setup(struct device *dev, u8_t options)
 	}
 
 	/* Activate all available options. Run in all cases. */
-	config.behaviour = NRF_WDT_BEHAVIOUR_RUN_SLEEP_HALT;
+	behaviour = NRF_WDT_BEHAVIOUR_RUN_SLEEP_HALT;
 
 	/* Deactivate running in sleep mode. */
 	if (options & WDT_OPT_PAUSE_IN_SLEEP) {
-		config.behaviour &= ~NRF_WDT_BEHAVIOUR_RUN_SLEEP;
+		behaviour &= ~NRF_WDT_BEHAVIOUR_RUN_SLEEP;
 	}
 
 	/* Deactivate running when debugger is attached. */
 	if (options & WDT_OPT_PAUSE_HALTED_BY_DBG) {
-		config.behaviour &= ~NRF_WDT_BEHAVIOUR_RUN_HALT;
+		behaviour &= ~NRF_WDT_BEHAVIOUR_RUN_HALT;
 	}
 
-	config.reload_value = m_timeout;
-
-	err_code = nrfx_wdt_init(&config, wdt_event_handler);
-	if (err_code != NRFX_SUCCESS) {
-		return -EBUSY;
-	}
-
-	IRQ_CONNECT(CONFIG_WDT_NRF_IRQ,
-		    CONFIG_WDT_NRF_IRQ_PRI, wdt_nrf_isr,
-		    DEVICE_GET(wdt_nrfx), 0);
-	irq_enable(CONFIG_WDT_NRF_IRQ);
+	nrf_wdt_behaviour_set(behaviour);
+	nrf_wdt_reload_value_set((m_timeout * 32768) / 1000);
```

Comment:
Add a comment like: "The watchdog timer is driven by the LFCLK clock running at 32768 Hz. The timeout value given in milliseconds needs to be converted here to watchdog ticks."